src/ansible_compat/config.py

```diff
@@ -95,10 +93,18 @@ class AnsibleConfig(_UserDict):  # pylint: disable=too-many-ancestors
         'COLLECTIONS_PATH': 'COLLECTIONS_PATHS',  # 2.10+ -> 2.9
     }
 
-    def __init__(self, config_dump: Optional[str] = None) -> None:
+    def __init__(
+        self,
+        config_dump: Optional[str] = None,
```

Comment:
Multiple ways of initializing the struct in the same initializer looks weird and mixes up multiple unrelated abstractions. It's best to do such things in custom classmethod initializers.

Also, I'd avoid any I/O in the default initializer for a similar reason. The main one should just accept the data and assign it internally.